service-runner.js

```diff
@@ -105,44 +101,96 @@ ServiceRunner.prototype._sanitizeConfig = function(conf, options) {
         if (/^\./.test(configFile)) {
             // resolve relative paths
             configFile = path.resolve(self._basePath + '/' + configFile);
         }
-        return fs.readFileAsync(configFile)
-        .then(function(yamlSource) {
-            self.config = self._sanitizeConfig(yaml.safeLoad(yamlSource),
-                    self.options);
-
-            // Make sure we have a sane config object by pulling in
-            // package.json info if necessary
-            var config = self.config;
-            config.package = package_json;
-            if (config.info) {
-                // for backwards compat
-                var pack = config.package;
-                pack.name = config.info.name || pack.name;
-                pack.description = config.info.description || pack.description;
-                pack.version = config.version || pack.version;
-            }
-        })
-        .catch(function(e) {
-            console.error('Error while reading config file: ' + e);
-            process.exit(1);
-        });
+        action = fs.readFileAsync(configFile);
     }
+
+    var package_json = {};
+    try {
+        package_json = require(self._basePath + '/' + 'package.json');
+    } catch (e) {}
+
+    return action.then(function(yamlSource) {
+        self.config = self._sanitizeConfig(yaml.safeLoad(yamlSource), self.options);
+        // Make sure we have a sane config object by pulling in
+        // package.json info if necessary
+        var config = self.config;
+        config.package = package_json;
+        if (config.info) {
+            // for backwards compat
+            var pack = config.package;
+            pack.name = config.info.name || pack.name;
+            pack.description = config.info.description || pack.description;
+            pack.version = config.version || pack.version;
+        }
+    })
+    .catch(function(e) {
+        console.error('Error while reading config file: ' + e);
+        process.exit(1);
+    });
+};
+
+/**
+ * Updates the config and sets instance properties.
+ *
+ * @param conf a configuration.
+ *             If undefined - config is loaded from config.yaml file
+ *             If Object - treated as already parsed configuration
+ *             If sting - treated a serialized yaml config
+ */
+ServiceRunner.prototype.updateConfig = function updateConfig(conf) {
+    var self = this;
+
+    return self.loadConfig(conf)
+    .then(function() {
+        var config = self.config;
+        var name = config.package && config.package.name || 'service-runner';
+
+        // display the version
+        if (self.options.displayVersion) {
+            console.log(name + ' ' + config.package.version);
+            process.exit(0);
+        }
+
+        // do we need to use Docker instead of starting normally ?
+        if (self.options.useDocker) {
+            self.options.basePath = self._basePath;
+            return docker(self.options, self.config);
+        }
```

Comment:
This and the previous block do not really belong here, the name of the method is `updateConfig`, but here other actions are taken. Need to move them out.